lib/Backends/Interpreter/InterpreterNodes.cpp

```diff
@@ -1473,25 +1473,128 @@ void BoundInterpreterFunction::fwdGatherRangesInst(
   }
 }
 
-void BoundInterpreterFunction::fwdScatterAssignInst(
-    const glow::ScatterAssignInst *I) {
+template <typename ElemTy>
+void BoundInterpreterFunction::fwdScatterDataInstCopyImpl(
+    const glow::ScatterDataInst *I) {
   Tensor *dataT = getTensor(I->getData());
   Tensor *indicesT = getTensor(I->getIndices());
   Tensor *slicesT = getTensor(I->getSlices());
 
-  size_t dataSliceSize =
-      dataT->size() / dataT->dims()[0] * dataT->getType().getElementSize();
+  assert(indicesT->dims().size() == 2 &&
+         "Index should be stored in 2D tensor!");
+  const size_t dataSliceSize =
+      slicesT->size() / slicesT->dims()[0] * dataT->getType().getElementSize();
```

Comment:
tiny nit: Even though they should be the same, I'd use `slicesT` element size instead since we're using its dims/size now just so it's not so confusing when reading this code.